Feature/roberta model

rahelbeloch opened the pull request:
@MatPrst  Could you have a look into this? This the setup for Roberta which I use to run some experiments on Lisa.

Commit:
Working on huggingface data preprocessing.

Commit:
Adding data script to get the data for the roberta baseline.

matprst committed:
Add huggingface tokenizer compatibility

matprst committed:
Update example notebook

Commit:
Merge branch 'main' into feature/roberta-model

Commit:
Working on Roberta Baseline.

Commit:
Working on Roberta Baseline.

Commit:
Working on Roberta Huggingface datasets.

Commit:
Merge pull request #4 from fvlieshout/feature/roberta-model-dataset

Dataloading compatible with huggingface tokenizer

Commit:
Merge remote-tracking branch 'origin/feature/roberta-model' into feature/roberta-model

# Conflicts:
#	model.py
#	train.py

Commit:
Working on Roberta Baseline.

Commit:
Working on Roberta Baseline.

Commit:
Working on Roberta Baseline.

Commit:
Making Roberta Lisa-ready

Commit:
Ignoring job files etc.

Commit:
Restructuring code, adding job_outputs to gitignore.

matprst commented on models/model.py:
```diff
@@ -0,0 +1,124 @@
+
+        self.model = DocumentClassifier(model_hparams)
+
+    def forward(self, inputs):
```
This is not being used as far as I can tell. It is similar to ```predict```. Maybe we should use forward instead?

rahelbeloch replied:
Yep, using forward, removing predict.

matprst commented on models/model.py:
```diff
@@ -0,0 +1,124 @@
+        return self.model(batch['input_ids'], batch['attention_mask']), batch['labels']
+
+    def count_parameters(self):
+        return sum(p.numel() for p in self.parameters() if p.requires_grad)
```
Is it possible that the number of parameters is incorrect since we set ```requires_grad = False``` for the first layers of the transformer?

rahelbeloch replied:
Yes. This method is currently not used anyways (I think the number of parameters is anyways automatically printed by the trainer).

rahelbeloch replied:
<img width="415" alt="Screenshot 2021-05-05 at 13 28 04" src="https://user-images.githubusercontent.com/8261902/117134330-bbfc1100-ada5-11eb-8502-4afaed6bd936.png">

matprst commented on models/model.py:
```diff
@@ -0,0 +1,124 @@
+
+    @staticmethod
+    def accuracy(predictions, labels):
+        # noinspection PyUnresolvedReferences
```
What is this comment referring to?

rahelbeloch replied:
Refers to this: 
![image](https://user-images.githubusercontent.com/8261902/117134133-77707580-ada5-11eb-9ab9-a5de3df173fd.png)

I don't really know why this warning is there in the first place.

matprst replied:
Ok, it looks like it's something related to Pycharm no?

matprst committed:
Add dataset parameter

Not sure if it's the best way to do it, but I think it makes it a bit more flexible.

matprst reviewed (commented):
Looks good! I added some comments and made a new commit adding ```dataset``` as a command line parameter. I'm not sure how to add the commit to this PR though.

Commit:
Working on PR review.

rahelbeloch commented:
@MatPrst I worked on your feedback and commented on the comments. The dataset parameter which you added is there! Pls check agaiin.

Commit:
Fixing the dataset parameter.

Commit:
Merge branch 'main' into feature/roberta-model

# Conflicts:
#	env.yaml

Commit:
Fixing merge conflicts in yaml
